rskj-core/src/main/java/co/rsk/net/discovery/message/NeighborsPeerMessage.java

```diff
@@ -46,6 +45,7 @@
 
     public NeighborsPeerMessage(byte[] wire, byte[] mdc, byte[] signature, byte[] type, byte[] data) {
         super(wire, mdc, signature, type, data);
+        this.nodes = new ArrayList<>();
```

Comment:
make nodes final

Reply:
This can be fixed but to arrange proper constructors it's needed to rework the whole PeerDiscoveryMessage hierarchy. I believe that it's better to address this on another PR